### ISSUE TITLE
Brann Bronzebeard + Master of Evolution

### ISSUE THREAD

**Issue by Muppet1** [demilich1/metastone#304]
Master of Evolution targeting another minion with Brann on board only triggered once.
It shd evolve the minion twice I think.

**Comment by webadict** [comment]
Yes, I believe that's correct... The issue is that this was changed due to how Recombobulator worked with Brann... If I change it back, I don't see any issues with it, but there may be something broken by doing so. I'll change it and we'll see.

**Comment by doctorpangloss** [comment]
There might be a few issues lurking with Brann. I think targeted battlecries just need to check that their targets are still on the board. I don't know if the most appropriate place is `canBeExecuted` for the `BattlecryAction` because that has more to do with conditions. We don't really want to repeat the action, so much as we want to repeat the spell with its target.

Otherwise you get weird effects like `ReturnMinionToHand` spell trying to move a minion in the set aside zone on the second evaluation of the battlecry.

I also believe battlecries that destroy the hero after the first battlecry create a really wonky situation on the second battlecry.

**Comment by doctorpangloss** [comment]
Is there a way for spells with user-selected targets to be aware when a transform has occurred? `TRANSFORM_REFERENCE` gets cleared after summoning, what about a `TARGET_OVERRIDE`?

**Comment by doctorpangloss** [comment]
I think this issue can be resolved if the second battlecry action checks that its target is still valid. This might look something like...

```java
final EntityReference target = battlecryAction.getSpell().hasPredefinedTarget() ? battlecryAction.getSpell().getTarget() : battlecryAction.getTargetKey();
final boolean targetable = target == null || target.isTargetGroup() ||
    /* The target is still among the valid targets for the action */
    getValidTargets(playerId, battlecryAction).stream().map(EntityReference::pointTo).anyMatch(er -> er.equals(target));
if (!battlecryAction.canBeExecuted(context, player) || !targetable) {
	return;
}
```
